eventstore_client.go

```diff
@@ -76,15 +92,139 @@ func getStats() (*stats, error) {
 		}
 	}
 
+	var parkedMessagesStatsResult []parkedMessagesStats
+	if enableParkedMessagesStats {
+		parkedMessagesStats, err := getSubscriptionParkedMessagesStats(subscriptionsStatsResult.result)
+		if err != nil {
+			log.WithError(err).Error("Error while getting parked messages for subscriptions.")
+		} else {
+			parkedMessagesStatsResult = *parkedMessagesStats
+		}
+	}
+
 	return &stats{
 		serverStatsResult.result,
 		gossipStatsResult.result,
 		projectionStatsResult.result,
 		infoResult.result,
 		subscriptionsStatsResult.result,
+		parkedMessagesStatsResult,
 	}, nil
 }
 
+func getSubscriptionParkedMessagesStats(subscriptions []byte) (*[]parkedMessagesStats, error) {
+	var result []parkedMessagesStats
+	jp.ArrayEach(subscriptions, func(jsonValue []byte, dataType jp.ValueType, offset int, err error) {
+		eventStreamID, _ := jp.GetString(jsonValue, "eventStreamId")
+		groupName, _ := jp.GetString(jsonValue, "groupName")
+
+		lastEventNumber, err := getParkedMessagesLastEventNumber(eventStreamID, groupName)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"eventStreamId": eventStreamID,
+				"groupName":     groupName,
+				"err":           err,
+			}).Warn("Error while getting parked messages last event number.")
+		}
+
+		truncateBeforeValue, err := getParkedMessagesTruncateBeforeValue(eventStreamID, groupName, lastEventNumber)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
```

Comment:
If the parked messages were never replayed, the `$tb` key is not present in metadata, and warning is logged here. This situation needs to be properly handled and oldest message metric needs to be calculated from the first message in the stream.

Reply:
The $tb value is defaulting to 0. As of that the following equation is always gonna be 0 
`lastEventNumber - totalNumberOfParkedMessages`  where  `TotalNumberOfParkedMessages := lastEventNumber - truncateBeforeValue`

And 0, is the first message in the stream.